Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -334,6 +331,11 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
             return nil
         }()
 
+        let purchasedProductsFetcher = OfflineCustomerInfoCreator.createPurchasedProductsFetcherIfAvailable(
+            diagnosticsTracker: diagnosticsTracker
+        )
+        let transactionFetcher = StoreKit2TransactionFetcher(diagnosticsTracker: diagnosticsTracker)
+
```

Comment:
Not really from this PR, but I'm curious. Since we're already creating the `StoreKit2TransactionFetcher` here, why not pass it to `OfflineCustomerInfoCreator.createPurchasedProductsFetcherIfAvailable` instead of creating another instance? Or do we want them to be two different instances for some reason?